### PR TITLE
minion: Wait for self to exist before syncing keys

### PR DESCRIPTION
**Pull request by kklin**
Should fix the CI errors.

### CODE REVIEW

File: minion/keys_test.go [hunk 1]
```diff
@@ -85,3 +86,22 @@ func TestSyncKeysError(t *testing.T) {
...
 	assert.EqualError(t, err, "operation not permitted")
 }
+
+func TestWaitForMinion(t *testing.T) {
+	t.Parallel()
+
+	start := time.Now()
+	conn := db.New()
+	go func() {
+		time.Sleep(1 * time.Second)
```
Comment (ejj) [inline minion/keys_test.go]: Meh, I don't like this much.  I'd prefer we either not unit test this function, or mock out the time.Sleep.  I generally don't like it if unit tests have to sleep

Comment (secant) [inline minion/keys_test.go]: Just wondering, why's that? (I'm assuming the tests are run in parallel, 1 second seems pretty short)

Comment (mmussomele) [inline minion/keys_test.go]: One thing you could do is run both functions in goroutines, and have them close channels after committing the minion and waiting for the minion, respectively. Then just have this at the end
```
select {
    case <-commitChan:
        break
    case <-waitChan:
        t.Fatal(...)
    case <- timeout:
        ....
}

Comment (mmussomele) [inline minion/keys_test.go]: actually what might be cleaner is to have each process send a known value into a channel, and assert that the values come out in the right order

Comment (ejj) [inline minion/keys_test.go]: Personally I think we should either (1) mock out time.Sleep (as done in some other tests) or (2) just delete the test.  It's only three lines of code that are "obviously correct", not that critical that its covered.

**Pull request merged**